terraform/modules/api-proxy-s3-sqs/api-gateway.tf

```diff
@@ -0,0 +1,110 @@
+resource "aws_api_gateway_account" "social_care_api_gateway_account" {
+  cloudwatch_role_arn = aws_iam_role.social_care_api_cloudwatch.arn
+}
+
+resource "aws_api_gateway_rest_api" "social_care_referrals_s3_api" {
+  name = "${var.resource_name_prefix}-s3-api"
+}
+
+resource "aws_api_gateway_resource" "form_submissions" {
+  rest_api_id = aws_api_gateway_rest_api.social_care_referrals_s3_api.id
+  parent_id   = aws_api_gateway_rest_api.social_care_referrals_s3_api.root_resource_id
+  path_part   = "form-submissions"
+}
+
+resource "aws_api_gateway_resource" "form_data" {
+  rest_api_id = aws_api_gateway_rest_api.social_care_referrals_s3_api.id
+  parent_id   = aws_api_gateway_resource.form_submissions.id
+  path_part   = "{form-data}"
+}
+
+resource "aws_api_gateway_method" "put_form_data" {
+  rest_api_id      = aws_api_gateway_rest_api.social_care_referrals_s3_api.id
+  resource_id      = aws_api_gateway_resource.form_data.id
+  http_method      = "PUT"
+  authorization    = "NONE"
+  api_key_required = true
+
+  request_parameters = {
+    "method.request.header.Accept"              = false
+    "method.request.header.Content-Type"        = false
+    "method.request.header.x-amz-meta-fileinfo" = false
+
+    "method.request.path.form-data" = true
+  }
+}
+
+resource "aws_api_gateway_method_response" "http_ok" {
+  rest_api_id = aws_api_gateway_rest_api.social_care_referrals_s3_api.id
+  resource_id = aws_api_gateway_resource.form_data.id
+  http_method = aws_api_gateway_method.put_form_data.http_method
+  status_code = 200
+}
+
+resource "aws_api_gateway_method_response" "http_bad_request" {
+  rest_api_id = aws_api_gateway_rest_api.social_care_referrals_s3_api.id
+  resource_id = aws_api_gateway_resource.form_data.id
+  http_method = aws_api_gateway_method.put_form_data.http_method
+  status_code = 400
+}
+
+resource "aws_api_gateway_method_response" "http_internal_server_error" {
+  rest_api_id = aws_api_gateway_rest_api.social_care_referrals_s3_api.id
+  resource_id = aws_api_gateway_resource.form_data.id
+  http_method = aws_api_gateway_method.put_form_data.http_method
+  status_code = 500
+}
+
+resource "aws_api_gateway_deployment" "social_care_referrals_s3_api_deployment" {
+  rest_api_id = aws_api_gateway_rest_api.social_care_referrals_s3_api.id
+  # Redeploy when there are new updates
+  triggers = {
+    redeployment = sha1(join(",", tolist([
```

Comment:
This is alright, though sometimes it's nicer to create several triggers for each thing that might change, that way you get visibility of which change actually hits the trigger. (Not required, just a nice to have)